spring-boot-starter-camunda-connectors/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java

```diff
@@ -55,170 +57,93 @@ public class InboundWebhookRestController {
-    if (HMACSwitchCustomerChoice.disabled
-        .name()
-        .equals(connectorProperties.getShouldValidateHmac())) {
-      return true;
-    }
-
-    HMACSignatureValidator validator =
-        new HMACSignatureValidator(
-            bodyAsByteArray,
-            headers,
-            connectorProperties.getHmacHeader(),
-            connectorProperties.getHmacSecret(),
-            HMACAlgoCustomerChoice.valueOf(connectorProperties.getHmacAlgorithm()));
-
-    return validator.isRequestValid();
+  private void incrementMetric(final String action) {
+    metricsRecorder.increase(METRIC_NAME_INBOUND_CONNECTOR, action, METRIC_WEBHOOK_VALUE);
   }
 
-  @Deprecated
-  private Map<String, Object> extractVariables(
-      WebhookConnectorProperties connectorProperties, Map<String, Object> context) {
-
-    String variableMapping = connectorProperties.getVariableMapping();
-    if (variableMapping == null) {
-      return context;
+  private Map<String, Object> assembleWebhookContextData(WebhookResponsePayload processedPayload) {
+    if (processedPayload == null) {
+      LOG.warn("Connector returned null");
+      return emptyMap();
     }
-    // Variable mapping is now supported on the Connector SDK level (see
-    // InboundConnectorProperties).
-    // We still support the old property for backwards compatibility.
-    LOG.warn(
-        "Usage of deprecated property `inbound.variableMapping`. "
-            + "Use `resultVariable` and `resultExpression` properties instead.");
-
-    return feelEngine.evaluate(variableMapping, context);
-  }
-
-  @Deprecated
-  private boolean activationConditionTriggered(
-      WebhookConnectorProperties connectorProperties, Map<String, Object> context) {
 
-    // at this point we assume secrets exist / had been specified
-    String activationCondition = connectorProperties.getActivationCondition();
-    if (activationCondition == null || activationCondition.trim().length() == 0) {
-      return true;
-    }
-    // Activation condition is now supported on the Connector SDK level (see
-    // InboundConnectorProperties).
-    // We still support the old property for backwards compatibility.
-    LOG.warn(
-        "Usage of deprecated property `inbound.activationCondition`. Use `activationCondition` instead.");
-    Object shouldActivate = feelEngine.evaluate(activationCondition, context);
-    return Boolean.TRUE.equals(shouldActivate);
+    return Map.of(
+        CONNECTOR_CTX_VAR_REQUEST,
+        Map.of(
+            CONNECTOR_CTX_VAR_BODY, Optional.ofNullable(processedPayload.body()).orElse(emptyMap()),
+            CONNECTOR_CTX_VAR_HEADERS,
+                Optional.ofNullable(processedPayload.headers()).orElse(emptyMap()),
+            CONNECTOR_CTX_VAR_PARAMS,
+                Optional.ofNullable(processedPayload.params()).orElse(emptyMap()),
+            CONNECTOR_CTX_VAR_CONNECTOR_DATA,
+                Optional.ofNullable(processedPayload.params()).orElse(emptyMap())));
```

Comment:
If this is a strictly defined data structure, can we use a record class here instead of assembling a map?

Reply:
Good point, let me try

Reply:
Done, I guess. Please take a look if that's what you thought it would look like. Thank you.

Reply:
I mean, I did it as a map in record class 😃 

But AFAIK when we correlate, FEEL wrapper works with maps.

Reply:
Wouldn't it work without wrapping it into a map? If I'm not mistaken, the FEEL wrapper can handle not only maps but any regular objects (haven't tried it with `record` classes though)

Reply:
I see what you mean. You say, that JSON mapper should de-serialize record as a map type reference.

I feel now that record class isn't really a great fit here. We still have to maintain structure `{request -> {body -> {}, headers -> {}}}`.

If you don't mind, I would like to return to regular map assembly in method, as it was before. IMO, record class in this case introduce more ambiguity. WDYT?

---

Also, on a side note, seems like we have sort of inconsistent behaviors. (1) When we [evaluate](https://github.com/camunda/connector-sdk/blob/main/runtime-util/src/main/java/io/camunda/connector/runtime/util/feel/FeelEngineWrapper.java#L135) an expression, we [put](https://github.com/camunda/connector-sdk/blob/main/runtime-util/src/main/java/io/camunda/connector/runtime/util/feel/FeelEngineWrapper.java#L84) `response` key. Which will be applicable for inbound as well. (2) We set `request` in runtime, and `response` in feel wrapper.

This is something we need to keep in mind, ideally rework before `1.x` release.

Reply:
Sure, makes sense - map assembly is fine then